diff --git a/comment_notify.migrate.inc b/comment_notify.migrate.inc
index 92b406f..08c1ad0 100644
--- a/comment_notify.migrate.inc
+++ b/comment_notify.migrate.inc
@@ -57,12 +57,65 @@ class CommentNotifyMigrationHandler extends MigrateDestinationHandler {
   }
 }
 
+/**
+ * Field handler.
+ */
+class CommentNotifyUserSettingsMigrationHandler extends MigrateDestinationHandler {
+  public function __construct() {
+    $this->registerTypes(array('user'));
+  }
+
+  /**
+   * Make the destination field visible.
+   */
+  public function fields() {
+    return array(
+      'node_notify' => t('Comment Notify: An integer indicating the default type of subscription: 0 means not subscribed, 1 means subscribed to all comments, and 2 means only subscribed to replies of this comment.'),
+      'comment_notify' => t('Comment Notify: An integer indicating the default type of subscription: 0 means not subscribed, 1 means subscribed to all comments, and 2 means only subscribed to replies of this comment.'),
+    );
+  }
+
+  /**
+   * Implements MigrateDestinationHandler::prepare().
+   *
+   * @param $user
+   *  The user object being prepared for saving.
+   * @param $row
+   *  Raw source data for the migration - ignored.
+   */
+  public function prepare($user, $row) {
+    // By default, set notifications off
+    if (!isset($user->node_notify)) {
+      $user->node_notify = 0;
+    }
+    if (!isset($user->comment_notify)) {
+      $user->comment_notify = 0;
+    }
+  }
+
+  /**
+   * Implements MigrateDestinationHandler::complete().
+   *
+   * @param $user
+   *  The user object that was just saved.
+   * @param $row
+   *  Raw source data for the migration - ignored.
+   */
+  public function complete($user, $row) {
+    comment_notify_set_user_notification_setting($user->uid, $user->node_notify, $user->comment_notify);
+  }
+}
+
 /*
  * Implementats hook_migrate_api().
  */
 function comment_notify_migrate_api() {
   $api = array(
     'api' => 2,
+    'destination handlers' => array(
+      'CommentNotifyMigrationHandler',
+      'CommentNotifyUserSettingsMigrationHandler',
+    ),
   );
   return $api;
 }
