diff --git a/components.inc b/components.inc
index 41cbdbf..a2e6dcf 100644
--- a/components.inc
+++ b/components.inc
@@ -74,6 +74,9 @@ class WebformComponents extends Migration {
     // from token.install, returns array('old' => 'new') without % prefix.
     if (strpos($row->value, '%') === 0) {
       switch($row->value) {
+        case '%username':
+          $row->value = '[current-user:name]';
+          break;
         case '%useremail':
           $row->value = '[current-user:mail]';
           break;
