diff --git a/d6/node.inc b/d6/node.inc
index 4c6d635..2452d73 100644
--- a/d6/node.inc
+++ b/d6/node.inc
@@ -116,6 +116,14 @@ class DrupalNode6Migration extends DrupalNodeMigration {
         }
       }
     }
+    // Join node_counter for Statistics support
+    if (Database::getConnection('default', $this->sourceConnection)
+        ->schema()->tableExists('node_counter')) {
+      $query->leftJoin('node_counter', 'nc', 'n.nid=nc.nid');
+      $query->addField('nc', 'daycount');
+      $query->addField('nc', 'timestamp');
+      $query->addField('nc', 'totalcount');
+    }
     return $query;
   }
 
